src/components/app-upload.js

```diff
@@ -35,27 +41,43 @@ class AppUpload extends Component {
     const value = e.target.value
     this.setState({ [name]: value })
   }
+
   handleSubmit = e => {
     e.preventDefault()
-    let { fileName, fileData, isPublic } = this.state
-    const { uploadFile } = this.props
-
+    let { fileName, fileData, isPublic, starred, size, extension } = this.state
+    const { uploadFile, currentFolder } = this.props
     if (fileName && fileData) {
-      var formData = new FormData()
-      fileData ? formData.append('upload', fileData) : void 0
-      formData.append('file_name', fileName)
-      formData.append('is_public', isPublic)
-      uploadFile(formData, this.successCallback)
+      let formdata = new FormData()
+      fileData ? formdata.append('upload', fileData) : void 0
+      formdata.append('file_name', fileName)
+      formdata.append('is_public', isPublic)
+      formdata.append('extension', extension)
+      formdata.append('starred', starred)
+      formdata.append('size', parseInt(size))
+      formdata.append('folder', parseInt(currentFolder.id))
+      uploadFile(formdata, this.successCallback)
     }
   }
-  handleCheck = () => {
+  handleCheckPublic = () => {
     this.setState({
       isPublic: !this.state.isPublic
     })
   }
+  handleCheckStar = () => {
+    this.setState({
+      starred: !this.state.starred
+    })
+  }
   successCallback = () => {
```

Comment:
you have written all the callback functions above as "handleSomthing" so rename this also as handleSuccess